bill/payment.go

```diff
@@ -319,6 +319,123 @@ func (pmt *Payment) calculate() error {
 	return nil
 }
 
+/** Billable interface implementation **/
+
+// GetSeries returns the series of the payment.
+func (pmt *Payment) GetSeries() cbc.Code {
+	return pmt.Series
+}
+
+// GetCode returns the code of the payment.
+func (pmt *Payment) GetCode() cbc.Code {
+	return pmt.Code
+}
+
+// GetIssueDate returns the issue date of the payment.
+func (pmt *Payment) GetIssueDate() cal.Date {
+	return pmt.IssueDate
+}
+
+// GetIssueTime returns the issue time of the payment.
+func (pmt *Payment) GetIssueTime() *cal.Time {
+	return pmt.IssueTime
+}
+
+// GetValueDate returns the value date of the payment.
+func (pmt *Payment) GetValueDate() *cal.Date {
+	return pmt.ValueDate
+}
+
+// GetTax returns the tax configuration of the payment.
+func (pmt *Payment) GetTax() *Tax {
+	return nil // no tax for payments
+}
+
+// GetPreceding returns the preceding document references of the payment.
+func (pmt *Payment) GetPreceding() []*org.DocumentRef {
+	return pmt.Preceding
+}
+
+// GetSupplier returns the supplier of the payment.
+func (pmt *Payment) GetSupplier() *org.Party {
+	return pmt.Supplier
+}
+
+// GetCustomer returns the customer of the payment.
+func (pmt *Payment) GetCustomer() *org.Party {
+	return pmt.Customer
+}
+
+// GetCurrency returns the currency of the payment.
+func (pmt *Payment) GetCurrency() currency.Code {
+	return pmt.Currency
+}
+
+// GetExchangeRates returns the exchange rates of the payment.
+func (pmt *Payment) GetExchangeRates() []*currency.ExchangeRate {
+	return pmt.ExchangeRates
+}
+
+// GetLines returns the lines of the payment.
+func (pmt *Payment) GetLines() []*Line {
+	return nil // no lines for payments
+}
+
+// GetDiscounts returns the discounts of the payment.
+func (pmt *Payment) GetDiscounts() []*Discount {
+	return nil // no discounts for payments
+}
+
+// GetCharges returns the charges of the payment.
+func (pmt *Payment) GetCharges() []*Charge {
+	return nil // no charges for payments
+}
+
+// GetPaymentDetails returns the payment details of the payment.
+func (pmt *Payment) GetPaymentDetails() *PaymentDetails {
+	return nil // no payment details for deliveries
+}
+
+// GetTotals returns the totals of the payment.
+func (pmt *Payment) GetTotals() *Totals {
+	return nil // no totals for payments
+}
```

Comment:
I don't think this is the best approach here. The `Payment` model does have some of these underlying properties with different types. Providing a getter that just returns nil because the types don't coincide could potentially be very confusing.